code/network/multiutil.cpp

```diff
@@ -1861,19 +1860,24 @@ int multi_can_message(net_player *p)
 
 	// only wing/team leaders can message
 	case MSO_SQUAD_LEADER:
+	{
 		// if the player has an invalid object #
 		if(p->m_player->objnum < 0){
 			return 0;
 		}
 
 		// check to see if he's a wingleader
-		sp = &Ships[Objects[p->m_player->objnum].instance];
-		if (sp->ship_name[strlen(sp->ship_name)-1] == '1') 
+		const ship_registry_entry* ship_regp = ship_registry_get(Ships[p->m_player->objnum].ship_name);
+		if (ship_regp == nullptr){
```

Comment:
This could be an Assertion because the entry should never be null here.

Reply:
I meant that the Assertion could take the place of the if() block, because the pointer should realistically *never* be null.  If it's ever null, there is a serious bug somewhere.  Thus it's not something that you need to normally check for.